lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -45,6 +50,38 @@ def __init__(self, config: PbenchServerConfig, logger: logging.Logger):
             role=API_OPERATION.READ,
         )
 
+    def get_paginated_obj(self, query: Query, json: JSON, url: str, page_limit: int):
+        """
+        Helper function to return a slice of dataset according to the page limit and
+        a page number and a json object containing next page url.
+        TODO: There are multiple ways to optimize the functionality
+            1. Use of caching mechanism to temporarily store the queried datasets
+            instead of querying the database with every page request.
+            2. Use of unique pointers to store the last returned row and then use
+            this pointer in subsequent page request instead of initial start to
+            narrow down the result.
+        """
+        paginated_result = {}
+        total_count = query.count()
+        if total_count > page_limit:
+            if "page_no" in json:
+                items = (
+                    query.order_by(Dataset.name)
+                    .limit(page_limit)
+                    .offset((json["page_no"] - 1) * page_limit)
+                    .all()
+                )
+                json["page_no"] += 1
+            else:
+                # This is the first page
+                items = query.limit(page_limit).all()
+                json["page_no"] = 1
```

Comment:
My instinct is that the only difference between the two values for `"page_no" in json` would be the value specified to `.offset()`.  That is, if the test is false, then the offset should be zero; but that doesn't mean that the query couldn't have provided an explicit zero-offset; analogously, is there any difference between not supplying the `.offset()` qualifier and supplying it with a default (presuably, zero) value?

So, this code should probably look more like this:
```
            offset_page = json.get("page_no", 1)
            items = (
                query.order_by(Dataset.name)
                .limit(page_limit)
                .offset((offset_page - 1) * page_limit)
                .all()
            )
            json["page_no"] = offset_page + 1
```